packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -209,11 +211,25 @@ export default class Proxy {
       xoaUpgradeTimeout
     )
 
-    const { id } = await this.registerProxy({
+    const props = {
       authenticationToken: proxyAuthenticationToken,
       name: this._generateDefaultProxyName(date),
       vmUuid: vm.uuid,
-    })
+    }
+
+    let id = proxyId
+    if (id !== undefined) {
+      const proxy = await this.getProxy(proxyId)
+      if (proxy.vmUuid !== undefined) {
+        app
+          .getXapi(proxy.vmUuid)
+          .deleteVm(proxy.vmUuid)
+          .catch(log.warn)
```

Comment:
Wouldn't it be better to delete the old VM first?

Reply:
Before the import of the new VM?

Reply:
Yes.

Reply:
So, this process will destroy the old VM event if the deployment of the new VM fails. And the user will finish without proxy VMs.